src/plugins/content_management/server/core/core.test.ts

```diff
@@ -0,0 +1,514 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+import { loggingSystemMock } from '@kbn/core/server/mocks';
+import type { ContentConfig } from './types';
+import { Core } from './core';
+import { createMemoryStorage, MockContent } from './mocks';
+import { ContentRegistry } from './registry';
+import { ContentCrud } from './crud';
+import type {
+  GetItemStart,
+  GetItemSuccess,
+  GetItemError,
+  CreateItemStart,
+  CreateItemSuccess,
+  CreateItemError,
+  UpdateItemStart,
+  UpdateItemSuccess,
+  UpdateItemError,
+  DeleteItemStart,
+  DeleteItemSuccess,
+  DeleteItemError,
+} from './event_types';
+import { EventBus } from './event_bus';
+
+const logger = loggingSystemMock.createLogger();
+
+describe('Content Core', () => {
+  describe('setup()', () => {
+    let core: Core;
+    let coreSetup: ReturnType<Core['setup']>;
+    let contentConfig: ContentConfig;
+
+    beforeEach(() => {
+      core = new Core({ logger });
+      coreSetup = core.setup();
+      contentConfig = {
+        storage: createMemoryStorage(),
+      };
+    });
```

Comment:
Nice that the whole content core can be spin up in a unit test!

Reply:
Yep, let's continue building that way with minimum dependencies...